music_assistant/providers/radioparadise/__init__.py

```diff
@@ -0,0 +1,513 @@
+        "stream_urls": {
+            "flac": "https://stream.radioparadise.com/beyond-flac",
+            "aac-320": "https://stream.radioparadise.com/beyond-320",
+            "mp3-192": "https://stream.radioparadise.com/beyond-192",
+            "aac-128": "https://stream.radioparadise.com/beyond-128",
+            "aac-64": "https://stream.radioparadise.com/beyond-64",
+        },
+        "api_url": "https://api.radioparadise.com/api/now_playing?chan=4",
+    },
+    "5": {
+        "name": "Radio Paradise - Serenity",
+        "description": "Don't panic, and don't forget your towel",
+        "stream_urls": {
+            "aac-128": "https://stream.radioparadise.com/serenity",
+        },
+        "api_url": "https://api.radioparadise.com/api/now_playing?chan=5",
+    },
+}
+
+# Stream format configurations
+BITRATE_FORMATS: dict[str, dict[str, ContentType]] = {
+    "flac": {"content_type": ContentType.FLAC},
+    "aac-320": {"content_type": ContentType.AAC},
+    "mp3-192": {"content_type": ContentType.MP3},
+    "aac-128": {"content_type": ContentType.AAC},
+    "aac-64": {"content_type": ContentType.AAC},
+}
+
+# Ordered list of formats for fallback logic
+FALLBACK_ORDER = ["flac", "aac-320", "mp3-192", "aac-128", "aac-64"]
+
+
+async def setup(
+    mass: MusicAssistant, manifest: ProviderManifest, config: ProviderConfig
+) -> ProviderInstanceType:
+    """Initialize provider(instance) with given configuration."""
+    return RadioParadiseProvider(mass, manifest, config)
+
+
+async def get_config_entries(
+    mass: MusicAssistant,  # noqa: ARG001
+    instance_id: str | None = None,  # noqa: ARG001
+    action: str | None = None,  # noqa: ARG001
+    values: dict[str, ConfigValueType] | None = None,  # noqa: ARG001
+) -> tuple[ConfigEntry, ...]:
+    """Return Config entries to setup this provider."""
+    return (
+        ConfigEntry(
+            key="stream_format",
+            type=ConfigEntryType.STRING,
+            label="Stream Quality",
+            description="Choose the audio quality/format for streams",
+            required=True,
+            default_value="flac",
+            options=[
+                ConfigValueOption(title="FLAC (Lossless)", value="flac"),
+                ConfigValueOption(title="AAC 320kbps", value="aac-320"),
+                ConfigValueOption(title="MP3 192kbps", value="mp3-192"),
+                ConfigValueOption(title="AAC 128kbps", value="aac-128"),
+                ConfigValueOption(title="AAC 64kbps", value="aac-64"),
+            ],
+        ),
+    )
+
+
+class RadioParadiseProvider(MusicProvider):
+    """Radio Paradise Music Provider for Music Assistant."""
+
+    def __init__(self, mass: MusicAssistant, manifest: ProviderManifest, config: ProviderConfig):
+        """Initialize the provider."""
+        super().__init__(mass, manifest, config)
+        self._channels_cache: dict[str, dict[str, Any]] = RADIO_PARADISE_CHANNELS.copy()
```

Comment:
as far as I can oversee you are not modifying any of the values in the global constant so I would skip this intermediate instance cache, just use the global const

Reply:
Done